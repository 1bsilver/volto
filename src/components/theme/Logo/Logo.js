/**
 * Logo component.
 * @module components/Logo
 */

import React from 'react';
import { Link } from 'react-router';

import LogoImage from './Logo.png';

/**
 * Logo component class.
 * @function Logo
 * @returns {string} Markup of the component.
 */
const Logo = () => (
  <Link id="portal-logo" title="Site" to="/">
    <img src={LogoImage} alt="Plone site" title="Plone site" />
  </Link>
);

export default Logo;
