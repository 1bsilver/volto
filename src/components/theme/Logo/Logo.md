Logo example source, with `intl` auto-injected.

```jsx static
<Logo />
```

Output:

```jsx noeditor
import {StaticRouter} from 'react-router-dom';

var IntlProvider = require('react-intl').IntlProvider;
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;

const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  actions: {},
  content: { get: {} },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <IntlProvider locale="en">
        <Logo />
      </IntlProvider>
    </StaticRouter>
  </Provider>
</div>;
```
