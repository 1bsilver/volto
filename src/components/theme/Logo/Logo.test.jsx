import React from 'react';
import renderer from 'react-test-renderer';
import Logo from './Logo';

test('renders a logo component', () => {
  const component = renderer.create(
    <Logo />
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
