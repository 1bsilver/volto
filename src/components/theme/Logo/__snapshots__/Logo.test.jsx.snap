exports[`test renders a logo component 1`] = `
<a
  id="portal-logo"
  onClick={[Function]}
  style={Object {}}
  title="Site">
  <img
    alt="Plone site"
    src={Object {}}
    title="Plone site" />
</a>
`;
